### ISSUE TITLE
Elasticsearch failed to initialize storage driver

### ISSUE THREAD

**Issue by andrew-jones** [google/cadvisor#1517]
Elastic have just released for GA Elasticsearch and Kibana 5.  The cadvisor elasticsearch driver doesn't seem to support the latest version of elasticsearch.

'Failed to initialize storage driver: failed to create the elasticsearch client - no Elasticsearch node available'

Changing my image tag from elasticsearch:latest to elasticsearch:2 has fixed the issue.  Is there a plan to make this work with 5.0?

**Comment by andrew-jones** [comment]
Should have looked, current requirement is 'gopkg.in/olivere/elastic.v2'.  v5 isn't stable yet.

**Comment by w-p** [comment]
Elastic is showing GA on v5, so I'd like to +1 the release.

**Comment by ppraj** [comment]
Yes , same issue , it fails on initiliazing . When can this be available

**Comment by GabKlein** [comment]
+1

**Comment by losoft** [comment]
+1

**Comment by mqasim1983** [comment]
+1

**Comment by sukki07** [comment]
+1

**Comment by dinoba** [comment]
+1

**Comment by se-jaeger** [comment]
+1

**Comment by entropeer-dev** [comment]
+1

**Comment by superbarne** [comment]
+1

**Comment by anderson-silva-0** [comment]
+1

**Comment by mpereira** [comment]
+1

**Comment by rpong79** [comment]
+1

**Comment by oopschen** [comment]
Check pull request 1597, hope will merge soon.

**Comment by slacksec** [comment]
+1

**Comment by disillusions** [comment]
+1

**Comment by OferE** [comment]
+1

**Comment by JinsYin** [comment]
+1

**Comment by man4j** [comment]
+1

**Comment by maheshveera** [comment]
+1

**Comment by dashpole** [comment]
ES5 support #1597 is currently blocked by #1458.

**Comment by sadok-f** [comment]
+1

**Comment by david-z-johnson** [comment]
+1

**Comment by vishalud** [comment]
+1

**Comment by johnmarcou** [comment]
+1

**Comment by selimekizoglu** [comment]
+1

**Comment by jjduhamel** [comment]
+1

**Comment by jarronshih** [comment]
+1

**Comment by shyd** [comment]
+1

**Comment by realcbb** [comment]
+1

**Comment by Enemchy** [comment]
+1

**Comment by raji90** [comment]
Does this issue still exist that it cadvisor works fine only with elasticserach version 2 and not above that? I have elasticserach instance with version 5.4.0 and on running cadvisor it gives me following error.

 docker run                                      \
>    --volume=/:/rootfs:ro                         \
>    --volume=/var/run:/var/run:rw                 \
>    --volume=/sys:/sys:ro                         \
>    --volume=/var/lib/docker/:/var/lib/docker:ro  \
>    --publish=9091:8080                           \
>    --detach=true                                 \
>    --name=cadvisor5                               \
>    docker.fmr.com/google/cadvisor:latest\
>    -storage_driver=elasticsearch -storage_driver_es_host="http://my_elk.fcorp.com:9200"

F0912 07:50:13.460296       1 cadvisor.go:117] Failed to initialize storage driver: failed to create the elasticsearch client - no Elasticsearch node available

**Comment by justinzyw** [comment]
+1

**Comment by nicolasgere** [comment]
+1

**Comment by ibc789** [comment]
+1

**Comment by cmoissl** [comment]
+1

**Comment by dims** [comment]
Closing this issue as it has been open since before January 1, 2024. If this is still relevant, please feel free to reopen or create a new issue.
